src/Hamiltonians/HOCartesian2BosonRelative.jl

```diff
@@ -0,0 +1,230 @@
+"""
+    log_abs_oscillator_zero(n)
+
+Compute the logarithm of the absolute value of the ``n^\\mathrm{th}`` 1D 
+harmonic oscillator function evaluated at the origin. The overall sign is
+determined when the matrix element is evaluated in [`ho_2brel_interaction`](@ref).
+"""
+function log_abs_oscillator_zero(n)
+    isodd(n) && return -Inf, 1.0
+    x, _ = SpecialFunctions.logabsgamma((1-n)/2)
+    y, _ = SpecialFunctions.logabsgamma(n+1)
+    result = log(pi)/4 + (n/2)*log(2) - x - y/2
+    return result
+end
+
+"""
+    ho_2brel_interaction(S, table, omm_i::OccupiedModeMap, omm_j::OccupiedModeMap)
+
+Returns the product of two one-dimensional harmonic oscillator functions 
+evaluated at the origin, 
+```math
+    v_{ij} = \\phi_i(0) \\phi_j(0)
+```
+Indices `i,j` start at `0` for the groundstate, must be even, and are bounded by
+the entries of `S` which defines the Cartesian HO modes.
+The values ``\\phi_i(0)`` are precomputed by [`HOCartesian2BosonRelative`](@ref) 
+and passed in as the vector `table`.
+The result is summed over all occupied modes in `omm_i` and `omm_j`.
+"""
+function ho_2brel_interaction(S, table, omm_i::OccupiedModeMap, omm_j::OccupiedModeMap)
+    states = CartesianIndices(S)
+    result = 0.0
+    # there should only be one particle but the sum is kept for consistency
+    for p_i in omm_i, p_j in omm_j
+        occ_i, mode_i = p_i
+        occ_j, mode_j = p_j
+        ho_indices = (Tuple(states[mode_i])..., Tuple(states[mode_j])...) .- 1
+        if all(iseven.(ho_indices))
+            doubly_evens = count(iseven.(ho_indices .÷ 2))
+            sign = (-1)^doubly_evens
+            result += sign * exp(sum(table[k÷2 + 1] for k in ho_indices)) * occ_i * occ_j
+        end
+    end
+    return result
+end
+
+"""
+    HOCartesian2BosonRelative(addr; S, Sx, ηs, g = 1.0, interaction_only = false)
```

Comment:
If you write the call signature like this, then it indicates that the keyword arguments `S, Sx, ηs`are all required, but they are not. In this case it is better to specify the call signature with the required positional argument only, and explain later (ideally as a list) which optional keyword arguments are available and what they do.
```suggestion
    HOCartesian2BosonRelative(addr; kwargs...)
```